test/sanitytest.c

```diff
@@ -76,12 +76,20 @@ static int test_sanity_unsigned_convertion(void)
     return 1;
 }
 
+static int test_sanity_pointer_sizes(void)
+{
+    if (!TEST_int_eq(sizeof(long), sizeof(void *)))
+        return 0;
+    return 1;
+}
+
 void register_tests(void)
 {
     ADD_TEST(test_sanity_null_zero);
     ADD_TEST(test_sanity_enum_size);
     ADD_TEST(test_sanity_twos_complement);
     ADD_TEST(test_sanity_sign);
     ADD_TEST(test_sanity_unsigned_convertion);
+    ADD_TEST(test_sanity_pointer_sizes);
```

Comment:
> how does #3191 work there?

Magically? Kidding. I suppose the question is how *did* LONG work [till it was replaced], right? I suppose it just so happens that Windows and VMS tolerate unaligned access. Note that I wrote "OS tolerate" and not "processor". Thing is that if processor doesn't tolerate unaligned access, vendor can still choose to handle misaligned access exception in kernel by looking at failing instruction a do what it intended to do byte by byte. And Windows does do exactly that. [Well, it doesn't have to on x86_64, but it does/did on Itanium.] And I bet VMS does it too. At least all other Alpha-based OSes do that. Problem with this approach is that even if OS hides it from you, you don't really want to have misaligned access on processors that don't tolerate it. Because it becomes *very* expensive from performance view-point. So expensive that you start wondering if it would be more honest to deliver signal, so that at least you know there is a problem...